src/modules/reducer.js

```diff
@@ -0,0 +1,61 @@
+const initialState = {
+  newId: 1,
+  reservationName: '',
+  reservationCategory: '',
+  reservationAddress: '',
+  reservations: [],
+};
```

Comment:
```suggestion
const initialReservation = {
  name: '',
  category: '',
  address: '',
};

const initailState = {
  newId: 1,
  reservations: [],
  reservation: initialReservation,
}
```

이렇게 한번 더 추상화 해준다면 코드 전반적으로 중복 로직이 제거될 것 같아요~